mapviz_plugins/src/robot_image_plugin.cpp

```diff
@@ -287,7 +295,21 @@ namespace mapviz_plugins
         texture_loaded_ = false;
       }
 
-      if (image_.load(filename_.c_str()))
+      std::string real_filename;
+      if (filename_.find("$(find ") != -1)
+      {
+        std::string package = filename_.substr(filename_.find("$(find ") + 7);
```

Comment:
What's the `+ 7` here for?  Is it safe to just got into an arbitrary index in the string like that?

Reply:
It's safe but apparently it can throw an exception. I'll revise it to check the range.

Reply:
Should be exception free now

Reply:
I think having at least a comment in there to explain where the +7 comes from would be helpful, or do a strlen() on that substring. Otherwise it's pretty hard to understand where that arbitrary number comes from.

Reply:
Also doesn't look like anything has changed, did you push?

Reply:
I apparently pushed to the wrong origin